src/ts/background/store/api-context/actions.ts

```diff
@@ -9,14 +10,14 @@ export const ACTION_TYPES = {
 
 type ApiWithProvider = { api: ApiPromise, provider: ProviderInterface }
 
-function doCreateApi (provider: ProviderInterface): Promise<ApiWithProvider> {
-  return ApiPromise.create(provider).then(api => ({ api, provider }))
+function doCreateApi (options: ApiOptions): Promise<ApiWithProvider> {
+  return ApiPromise.create(options).then(api => ({ api, provider: options.provider!! }))
```

Comment:
should we make provider optional in response?

Reply:
yes, it's optional now